### DISCUSSION TITLE
Scheduling auto update for systems or groups

### DISCUSSION THREAD

**Discussion by gabardyne** [uyuni-project/uyuni#4552] [Q&A] [answered]
Hello,

I'm wondering if it is possible to schedule a patch installation for a system or a system group, for example every friday at 11PM, in Uyuni.

I've tried several things without success, and I'm kind of lost...

Regards,

Paul

**Comment by vega82** [comment, marked as answer]
+1  this is a indispensable feature

**Reply by dvosburg** [reply]
Create a Configuration State Channel called 'update_eveything'

This is the state to accomplish this:

```
update_pkg:
  pkg.uptodate:
    - refresh : True
    
int_reboot_if_needed:
  cmd.run:
    - name: shutdown -r +5
{%- if grains['os_family'] == 'RedHat' and grains['osmajorrelease'] >= 8 %}
    - onlyif: 'dnf -q needs-restarting -r; [ $? -eq 1 ]'
{%- elif grains['os_family'] == 'RedHat' and grains['osmajorrelease'] <= 7 %}
    - onlyif: 'needs-restarting -r; [ $? -eq 1 ]'
{%- elif grains['os_family'] == 'Debian' %}
    - onlyif: 
      - test -e /var/run/reboot-required
{%- elif grains['os_family'] == 'Suse' and grains['osmajorrelease'] <= 12 %}
    - onlyif:
      - test -e /boot/do_purge_kernels
{%- else %}
    - onlyif: 'zypper ps -s; [ $? -eq 102 ]'
{%- endif %}
```

You can then assign this Configuration Channel to a System Group, or even an individual system, and schedule a recurring highstate.
